src/test/java/org/takes/facets/auth/PsByFlagTest.java

```diff
@@ -46,11 +53,64 @@ public void skipsIfNothingFound() throws IOException {
         MatcherAssert.assertThat(
             new PsByFlag(
                 new PsByFlag.Pair(
-                    "some-key", new PsFake(true)
+                    "test", new PsFake(true)
                 )
-            ).enter(new RqFake("GET", "/?PsByFlag=x")).hasNext(),
+            ).enter(
+                new RqFake("GET", "/?PsByFlag=x")
+            ).hasNext(),
             Matchers.is(false)
         );
     }
 
+    /**
+     * PsByFlag finds flag and authenticates user.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void flagIsFoundUserAuthenticated() throws IOException {
+        MatcherAssert.assertThat(
+            new PsByFlag(
+                new PsByFlag.Pair(
+                    "some-key", new PsFake(true)
+                )
+            ).enter(
+                new RqFake("POST", "/?PsByFlag=some-key")
+            ).next().urn(),
+            Matchers.is("urn:test:1")
+        );
+    }
+
+    /**
+     * PsByFlag wraps response with authenticated user.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void exitTest() throws IOException {
+        final Response response = new RsWithStatus(
+            new RsWithType(
+                new RsWithBody("<html>This is test response</html>"),
+                    "text/html"
+            ),
+            200
+        );
+        MatcherAssert.assertThat(
+            new PsByFlag(
+                ImmutableMap.of(
+                    "key", (Pass) new PsFake(true)
```

Comment:
@yegor256 Because maps are with different types ```ImmutableMap.of("some-key", new PsFake(true));``` - this is ```Map<String, PsFake>``` map, but constructor is using ```Map<String, Pass>``` map. I know that custing  is not a good practice, but here I've used it to workaround limit of the API. I create object in the same method where I cast and we can be 100% sure that this cast will work and also casting to the parent type works implicitly. Also we can change constructor here to:
```Java
public PsByFlag(final Map<String, ? extends Pass> map) {
        this(PsByFlag.class.getSimpleName(), map);
    }
```